added hwinfo

bobvanderlinden opened the pull request:
This adds `hwinfo` and its depedency `libx86emu` from OpenSUSE.

bobvanderlinden committed:
added hwinfo

edolstra commented on pkgs/development/libraries/libx86emu/default.nix:
```diff
@@ -0,0 +1,39 @@
+  '';
+
+  meta = with stdenv.lib; {
+    description = "x86 emulation library.";
```
No period please.

edolstra commented on pkgs/tools/system/hwinfo/default.nix:
```diff
@@ -0,0 +1,39 @@
+  buildInputs = [ rpmextract libx86emu flex ];
+
+  meta = with stdenv.lib; {
+    description = "Hardware detection tool from openSUSE.";
```
Idem.

edolstra commented on pkgs/development/libraries/libx86emu/default.nix:
```diff
@@ -0,0 +1,39 @@
+  meta = with stdenv.lib; {
+    description = "x86 emulation library.";
+    license = licenses.bsd;
+    homePage = http://download.opensuse.org/source/factory/repo/oss/suse/src/;
```
This should be `homepage`.

bobvanderlinden commented:
I've removed the periods in the descriptions and changed homePage to homepage. I copied this from a different package, so there are still others that use the wrong convention. Let me know if it's helpful to fix these as well.

codyopel commented:
It would probably be easier to pull a tarball from github than building from an rpm.
https://github.com/openSUSE/hwinfo
https://github.com/wfeldt/libx86emu

bobvanderlinden commented:
Hmm, I used the same process ArchLinux uses for their packages. I'll give the github repos a go.

codyopel commented:
Yeah, for some strange reason it looks like gentoo and arch at least are using the rpm's.  You can easily switch fetchurl to use the tarball from the release though. (e.g. https://github.com/openSUSE/hwinfo/archive/21.12.tar.gz)  It just makes it a bit cleaner not having to use rpmextract and then you won't need the ```unpackPhase``` either.

bobvanderlinden commented:
Unfortunately they do some magic with git to create a changelog and know the version, which fails when using .tar.gz. I've fixed the version issue, but not the changelog yet. I'll give this another look later tonight (probably use the git repo, instead of .tar.gz).

bobvanderlinden commented:
Ok, the current version fetches the .tar.gz files from github. I did need to create a VERSION file and needed to skip generating the changelog (which did not seem to be used by the compiling process nor executable itself).

Thanks @codyopel for the suggestion. It's much cleaner now

lethalman commented:
Travis passed, thanks.
